### ISSUE TITLE
Crash when fitting a Structure

### ISSUE THREAD

**Issue by steven-tk** [EVEIPH/EVE-IPH#122]
An Unhandled Exception has occured and EVE Isk per Hour will now close.

Please fill out the following information so I can reproduce the bug

**What is your Operating System?** 
I run IPH via Crossover 17.1 (Wine Wrapper / WinXP) on MacOs 10.13.6

**What tab or screen did the error occur?**
In the structure fittings window trying to add fittings via double-click.

**What are the steps to reproduce the Error?**
- Go to manufacturing
- Switch the default station to structure (Sotiyo)
- Click edit fitting
- No way to fit it (missing icons? See SS)
- Tried double clicking on the fittings names (tried both service modules and multiple rigs - always crashes instantly)

**Web link to a screenshot of your error:** 
http://snpy.in/ais4Ue

In addition to a screenshot, copy the data below and send to developer.

Source: EVE Isk per Hour
Message: Object reference not set to an instance of an object.
Raw Error Text: System.NullReferenceException: Object reference not set to an instance of an object.
   at EVE_Isk_per_Hour.frmUpwellStructureFitting.LoadImageInFreeSlot(Int32 ModuleTypeID, String ModuleName, String GroupTag, String GroupName)
   at EVE_Isk_per_Hour.frmUpwellStructureFitting.LoadSelectedImageInFreeSlot()
   at EVE_Isk_per_Hour.frmUpwellStructureFitting.ServiceModuleListView_ItemActivate(Object sender, EventArgs e)
   at System.Windows.Forms.ListView.OnItemActivate(EventArgs e)
   at System.Windows.Forms.ListView.WmReflectNotify(Message& m)
   at System.Windows.Forms.ListView.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
Error Tracking: 

---

If you need anything else let me know!

Edit:
As a workaround: is there a way to somewhere manually make a fit by editing a text file?

**Comment by Zifrian** [comment]
I'll fix this from crashing but you need to have the images downloaded and in the folder where the program is run to show them. I'm not sure how that works with wine.

**Comment by steven-tk** [comment]
I'm surprised the pictures don't come with the regular binary file download?
Are they missing or is it a bug ?

Wine just emulates (well technically it's not an emulator) the necessary windows parts needed for the software. In terms of file structure it behaves like regular windows.
So I just need to know where the pictures would be placed in a normal installation and where to download them.

**Comment by Zifrian** [comment]
They are in the binary. All stored in the EVEIPH Images folder.

**Comment by Zifrian** [comment]
Let me know if you found the images. Closing issue for now.

**Comment by steven-tk** [comment]
The Images are there, just not in a folder. Given how they are labeled (e.g. "EVEIPH Images\684_6.png") something went wrong when de-compressing and it was fixed using a different tool for that.
It might be worth mentioning somewhere that the macOS system default is bugged that way and recommend e.g. https://theunarchiver.com/ to fix that.

With the pictures now loading it also fixed the crash for me when trying to fit equipment to a structure which now works fine. (Yay!)

Additionally this seemed to have fixed the issue of a very long time required to update the prices which now is significantly faster (10-15min rather than 45min+ via CCP Data) and doesn't get stuck at the end anymore like it did most times.

I haven't installed the new update yet as the updater seemed to crash every time: http://snpy.in/AlHEuK
I can create a new issue for that if you prefer?

**Comment by Zifrian** [comment]
Thanks, the new version has some fixes for update prices as well. Speed is still going to be an issue I think but I can do a full download in about 4.5 minutes from CCP.

The updater was crashing, but I just fixed it. The issue was if you installed on another drive rather than the one you run Windows on. Your screenshot looks like it is an issue with a missing library however. 

I use the one from .Net. It's System.IO.Compression.FileSystem.dll in .Net 4.6.1 I only use it in the updater to unpack the zipped file for the images. 

Let me know if you need more.

**Comment by steven-tk** [comment]
I created a new bottle in wine running win7 and installed .Net 4.6.1
With that not only did the updater run flawlessly but so did the installer - so far I always had to use the binary files. If I remember correctly that was the suggested approach in your old forum post (granted that was years ago)
I might be blind at times so chances are I missed it, but I never saw it mentioned anywhere you need .Net 4.6.1 and therefore windows 7 to run it - might be nice mentioning that somewhere especially for people wanting to run it via Wine. 

In fact there is a pre-made install profile for wine and its commercial brother crossover but they both are way out of date. If I got too much time in the coming days I'll check if I can update it. (Already applied for access). Once that is done you can advertise it as running on Mac and Linux :)

**Comment by Zifrian** [comment]
OK thanks for the help.
